### ISSUE TITLE
HCL build source matching not working and naming schema elusive

### ISSUE THREAD

**Issue by timblaktu** [hashicorp/packer#10902]
Similar to #10677, I'm having trouble with the `only` concept again, but now using `-only` from packer build cli, to limit which builders get run. I'm using 1.7.0 and HCL.

For these sources:

```
source "virtualbox-iso" "debian10" {                                                                                                                                                                                                                                                        
  ...
}

source "vsphere-iso" "debian10" {
  ...
}
```

..and builders which use the above sources (not I override the `name` is the virtualbox-iso sources)

```
build {
  name = "debian10"
  source "source.vsphere-iso.debian10" {
  ...
  }
  source "source.virtualbox-iso.debian10" {
    name                    = "debian10-headless"
    vm_name                 = "{{build_name}}-headless-${local.scm_suffix}"
  }
  source "source.virtualbox-iso.debian10" {
    name                    = "debian10-gnome"
    vm_name                 = "{{build_name}}-gnome-${local.scm_suffix}"
  }
  .
  .
}
```

I'm trying to invoke, e.g.. just the single vsphere-iso builder and the headless virtualbox-iso variant, I try:

```
packer build -force -only 'vsphere-iso.debian10,virtualbox-iso.debian10-headless' -var output_directory_root=artifacts .
```

But it doesn't match and nothing gets built.

As over in #10677, I'm still "overriding" the name field inside my virtualbox-iso build sources, so per @SwampDragons comment, I'm trying to use my overridden names, but obviously I'm not doing it right and the HCL build/source naming schema continues to elude me. I assumed this is cockpit error so I made it a questions instead of a bug.

**Comment by azr** [comment]
Hello there, thanks for opening, I agree, I think that #11050 should help with the issue here ? Tell me what you think.

**Comment by timblaktu** [comment]
Yes, that looks very helpful @azr. I'd like to try it out in my CI environment. I assume this will be included in 1.7.3. Any ETA on the next release?

**Comment by azr** [comment]
Yes for 1.7.3, you could also try it with the binary of the feature branch: https://app.circleci.com/pipelines/github/hashicorp/packer/10305/workflows/578288c5-b128-4ea8-835f-3e5e34d5fbdc/jobs/125294

**Comment by timblaktu** [comment]
Thanks @azr. I do pretty much all of my Packer work on Jenkins agents that are provisioned using [this ansible role](https://github.com/sbaerlocher/ansible.virtualization/tree/0.0.10/roles/packer) which [installs Packer by expanding a zip file](https://github.com/sbaerlocher/ansible.virtualization/blob/98b21fcb1b3c6ece282469f360ad7fef58f4b8f0/roles/packer/tasks/distribution/Linux.yml#L7) found at this URL:

```https://releases.hashicorp.com/packer/{{ packer_version }}/packer_{{ packer_version }}_linux_amd64.zip```

Is it safe for me to assume the format of the zip file you archive at the circleci job you references, i.e. this:

```https://125294-8966356-gh.circle-artifacts.com/0/packer_linux_amd64.zip```

is the same as those at https://releases.hashicorp.com/packer? If so, I could easily test the new version by forking the role and extending it to be able to pull the zip from an arbitrary URL.

Where are the [nightly release which is available via Github Releases](https://github.com/hashicorp/packer/releases/tag/nightly) built from? If that contains the commit you referenced, then I could just refer to it.

**Comment by azr** [comment]
The CircleCI artifact URL is rather random but using it is okay, CircleCI provides no guarantee regarding the time the binary will be available though, and that's about all I know about CC artifacts, I think.

Nightly releases also are built in CircleCI (official packer bins too). Nightly builds are then uploaded to the GitHub nightly release. The nightly build also going to contain the fix ! 🙂

**Comment by chien-splunk** [comment]
I had a similar problem trying to get `only` to work on 1.7.2 and the error I had was that I did not understand the naming used by the `only` option. 

The [docs](https://www.packer.io/docs/commands/build#only-foo-bar-baz) is not clear on the naming convention, it seems to be `<build name>.<source type>.<source name>`. In your case, I think it'll match if you do `-only debian10.vsphere-iso. debian10`. If you run `packer build -debug .` and ctrl-c out, it also lists the sources found and their names, which is how I found the proper naming.
